agent/grpc-external/services/peerstream/stream_resources.go

```diff
@@ -286,6 +319,11 @@ func (s *Server) HandleStream(streamReq HandleStreamRequest) error {
 
 			return nil
 
+		// We haven't received a heartbeat within the expected interval. Kill the stream.
+		case <-incomingHeartbeatCtx.Done():
+			logger.Error("ending stream due to heartbeat timeout")
+			return fmt.Errorf("heartbeat timeout")
```

Comment:
The caller of `HandleStream` will retry on error; I'm assuming heartbeat timeout is something we consider retryable?

Reply:
Sounds good, thanks for thinking this through!